app/assets/javascripts/modules/inline-image-modal.js

```diff
@@ -0,0 +1,79 @@
+/* global MultiSectionViewer */
+
+function InlineImageModal (trigger) {
+  this.$trigger = trigger
+  this.$modal = document.getElementById('modal')
+}
+
+InlineImageModal.prototype.init = function () {
+  var $module = this
+
+  if (!this.$trigger || !this.$modal) {
+    return
+  }
+
+  this.$multiSectionViewer = new MultiSectionViewer(this.$modal.querySelector('[data-module="multi-section-viewer"]'))
+  this.$multiSectionViewer.init()
+
+  this.$trigger.addEventListener('click', function (event) {
+    event.preventDefault()
+    $module.performAction($module.$trigger)
+  })
+}
+
+InlineImageModal.prototype.fetchModalContent = function (url) {
+  var controller = new window.AbortController()
+  var headers = { 'X-Requested-With': 'XMLHttpRequest' }
```

Comment:
As we chatted about the other day - since this isn't actually an XMLHttpRequest maybe we could go with a header that specifies context?

Reply:
We could go with `X-Context` if you like, and this would be blank by default?

Reply:
Yup referencing context sounds good.

It's probably worth checking the guidance on custom HTTP headers, from a quick search I think we're discouraged from using the X prefix on custom. Might also be prudent to include GOVUK or Content Publisher to make it really clearly custom.

Reply:
@kevindew, from my quick reading, the `X-` prefix is discouraged except for internal uses that will definitely remain internal, which I think is true here. However, `Content-Publisher-Context` sounds good too, so I think I'll go with that.

Reply:
Done :-). Normally I might avoid putting it in ApplicationController straight away, but I've already done the next PR that demonstrates that it's used multiple times.